docs/semgrep-code/remove-duplicates.md

```diff
@@ -5,26 +5,116 @@ title: Remove duplicate findings
+
+### `match_based_id`
+
+While the `syntactic_id` associated with a finding helps track findings within the same file, there are times when the way Semgrep generates the `syntactic_id` isn't sufficient for the identification and deduplication of findings. 
+
+Semgrep generates the `match_based_id` for a finding using the following information for a finding:
+
+- The file path
+- The name of the rule that generated the finding
+- The rule pattern with the metavariables' values substituted in
+
+This information is combined and then hashed. At this point, Semgrep appends the **index**, a value generated by determining the number of times the rule involved matched code in the file. Note that the index is appended to the hash, not combined with the other finding information before hashing. This is done to preserve information on how findings are related. For example, `finding0` with `match_based_id = 123_0` and `finding1` with `match_based_id = 123_0` indicate that both were generated from the same rule matching the same code in the same file.
+
+Because Semgrep uses the rule pattern instead of the literal code syntax to generate `match_based_id`, code changes that do not impact the overall function don't hinder Semgrep's ability to recognize that the finding isn't a duplicate of an existing finding.
+
+![Semgrep AppSec Platform groups together findings on different branches](/img/matched-findings.png)
+_**Figure**. Semgrep AppSec Platform groups two findings on separate branches that are duplicates._
+
+For example, if the original file scanned is:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` is:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+If the following change is made to the original file:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+c_1 = 5
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` does not change:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+This means that the `match_based_id` itself doesn't change, allowing Semgrep to identify that the two findings are the same, allowing deduplication. Furthermore, this process enables Semgrep to ignore lines that do not impact code function. For example, the addition of comments to a code file would create a new `syntactic_id`, but not a new `match_based_id`.
+
+### How Semgrep AppSec Platform handles findings using `syntactic_id` and `match_based_id`
```

Comment:
I'm not convinced we should be this detailed here. The actual interaction of syntactic ID and match-based ID in this context still confuses even me (for example, I've been told there are cases where we still use syntactic ID for certain things that I don't think fall into item 1)

Reply:
I think I would agree with Alexis's comment here as well (sorry I know I mentioned that most of the information in the doc was fine to expose). I think we can most likely just push `match_based_id` as the hash that we use to deduplicate while `syntactic_id` is another hash that we use more for internal debugging purposes because we don't store code (of course unless the user gives us code access to view their code).